sys/hoon.hoon

```diff
@@ -4264,296 +4270,477 @@
+::  a rune, an optional tall-form terminator, optionally a short-form (if
+::  you don't supply a short-form, it'll just construct the standard
+::  wide-form (e.g. "?~(x x ~)") for you, and a list of sub-plums.
+::
+++  rune
+  |=  $:  rune=cord
+          term=(unit cord)
+          short=(unit [cord cord cord])
+          kids=(list plum)
+      ==
+  ^.  plum
+  |^  :-  %sbrk
+      :+  %tree
+        :-  (rune-wide-form rune short)
+        ?~  term  (tall-fixed rune)
+        (tall-running rune '' u.term)
+      kids
+  ::
+  ::  If you just give this a rune, it'll build the standard wide-form.
+  ::  Otherwise, it'll just ose the one that you gave it.
+  ::
+  ++  rune-wide-form
+    |=  [rune=cord short=(unit [fst=cord mid=cord lst=cord])]
+    ^-  (unit (pair cord (unit [cord cord])))
+    =*  fst  (cat 3 rune '(')
+    =*  std  `[' ' `[fst ')']]
+    ?~  short  std
+    `[mid.u.short `[fst.u.short lst.u.short]]
+  --
+::
+::  Just a helper function for constructing a wide-form %tree plum.
+::
+++  simple-wide
+  |=  [init=cord sep=cord end=cord kids=(list plum)]
+  ^-  plum
+  =/  fmt=plumfmt  [wide=[~ sep [~ init end]] tall=~]
+  [%tree fmt kids]
+::
+::  Convenience function that builds a plum for a subexpression. The
+::  `%sbrk` tells the pretty-printer that this is a valid place to
+::  switch from tall mode to wide mode.
+::
+++  subtree
+  |=  [p=plumfmt q=(list plum)]
+  ^-  plum
+  [%sbrk [%tree p q]]
+::
+::  Convenience for generating plums that look like s-expressions. Useful
+::  for quickly getting decent-looking debug output.
+::
+++  sexp
+  |=  [sym=cord kids=(list plum)]
+  ^-  plum
+  =/  head=cord     (cat 3 '(' sym)
+  =/  headspc=cord  (cat 3 head ' ')
+  =/  symcol=cord  (cat 3 sym ':')
+  =/  fmt=plumfmt   [[~ ' ' [~ headspc ')']] [~ symcol [~ '' '']]]
+  ?~  kids  (cat 3 '(' (cat 3 sym ')'))
+  [%sbrk [%tree fmt kids]]
+::
+::  --
+::
 ::  highly unsatisfactory temporary tank printer
 ::
 ++  plum-to-tank
   |=  =plum
   ^-  tank
-  ?@  plum  [%leaf (trip plum)]
-  ?-  -.plum
-    %|  :+  %rose
-          ["" " " ""]
-        (turn list.plum |=(@ta [%leaf (trip +<)]))
-    %&  =/  list  (turn list.plum ..$)
-        ?~  tall.plum
-          ?>  ?=(^ wide.plum)
-          =?  enclose.u.wide.plum  ?=(~ enclose.u.wide.plum)  `['{' '}']
-          :+  %rose
-            :*  (trip delimit.u.wide.plum)
-                (trip +<:enclose.u.wide.plum)
-                (trip +>:enclose.u.wide.plum)
-            ==
-          list
-        ?:  ?=(^ indef.u.tall.plum)
-          :+  %rose
-            :*  (trip sigil.u.indef.u.tall.plum)
-                (weld (trip intro.u.tall.plum) "[")
-                (weld "]" (trip final.u.indef.u.tall.plum))
-            ==
-          list
-        :+  %palm
-          :*  (weld (trip intro.u.tall.plum) "(")
-              ""
-              ""
-              ")"
-          ==
-        list
-  ==
+  [%leaf "TANK"]
+  ::  ?@  plum  [%leaf (trip plum)]
```

Comment:
Can you include a description of why this code is commented out?

Reply:
It's dead code.